### PR TITLE
LibWeb/CSS: Use TokenStream in more places

### PR DESCRIPTION
**Pull request by AtkinsSJ**
The end goal is to consistently use TokenStream everywhere, so that mistakes are harder to make and easier to spot. (Such as not consuming all tokens, or skipping one accidentally.) The ones here aren't that exciting, but there are some harder ones left and my brain wasn't up to those today. :sweat_smile: 

And as a tangent, split up the unwieldy `convert_to_rule()` function into one per rule, and add some documentation to them.

### CODE REVIEW

File: Userland/Libraries/LibWeb/CSS/Parser/Parser.h [hunk 1]
```diff
@@ -222,6 +222,10 @@ class Parser {
 
     CSSRule* convert_to_rule(NonnullRefPtr<Rule>);
     CSSMediaRule* convert_to_media_rule(NonnullRefPtr<Rule>);
+    JS::GCPtr<CSSKeyframesRule> convert_to_keyframes_rule(Rule&);
+    JS::GCPtr<CSSImportRule> convert_to_import_rule(Rule&);
+    JS::GCPtr<CSSNamespaceRule> convert_to_namespace_rule(Rule&);
+    JS::GCPtr<CSSSupportsRule> convert_to_supports_rule(Rule&);
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/CSS/Parser/Parser.h:228]: Not sure about the return types here. The existing function return raw pointers, but the `CSSFooRule::create()` functions all return a `NonnullGCPtr`, so it seemed weird to not just return that? This code is a couple of years old though so we might not have had `GCPtr` and friends around back then.

Comment (awesomekling) [reply Userland/Libraries/LibWeb/CSS/Parser/Parser.h:228]: Yeah `GCPtr` or `NonnullGCPtr` is nicer IMO, communicates better what's being returned.

Comment (AtkinsSJ) [reply Userland/Libraries/LibWeb/CSS/Parser/Parser.h:228]: OK, I'll update the existing ones too then. :+1:

**Comment by AtkinsSJ** [comment]
**Changes:**
- Make the existing CSSRule parsing methods return GCPtrs
- Put the debug logging I added behind CSS_PARSER_DEBUG, like I meant to. :sweat_smile:

**Pull request merged**